src/ThreeEditor/ThreeEditor.tsx

```diff
@@ -36,18 +53,77 @@ function ThreeEditor(props: ThreeEditorProps) {
 			editor?.signals.sceneGraphChanged.dispatch();
 		} else containerEl.current?.blur();
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [props.focus]);
+	}, [props.focus, editor]);
+
+	useEffect(
+		() => {
+			editor?.signals.windowResize.dispatch();
+		},
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[editor, props.sidebarProps]
+	);
 
 	return (
-		<div
-			className='ThreeEditor'
-			ref={containerEl}
-			style={{
-				position: 'relative',
+		<Box
+			sx={{
+				width: '100%',
+				height: '100vh',
 				display: 'flex',
-				flexGrow: 1
-			}}
-		/>
+				flexDirection: 'row',
+				overflow: 'hidden'
+			}}>
+			<Box
+				sx={{
+					display: 'flex',
+					flexDirection: 'column',
+					flexGrow: 1
+				}}>
+				<EditorAppBar editor={editor} />
+				<EditorMenu editor={editor} />
+				<div
+					className='ThreeEditor'
+					ref={containerEl}
+					style={{
+						position: 'relative',
+						display: 'flex',
+						flexGrow: 1
+					}}>
+					{!editor && (
+						<CircularProgress
+							sx={{
+								margin: 'auto'
+							}}
+						/>
+					)}
+				</div>
+			</Box>
+
+			<AppBar
+				position='static'
+				color='secondary'
+				sx={{
+					'width': 350,
+					'&.MuiAppBar-colorSecondary': {
+						backgroundColor:
+							theme.palette.mode === 'dark'
+								? theme.palette.secondary.dark
+								: theme.palette.secondary.light
+					}
+				}}>
+				<Typography
+					sx={{
+						padding: 2
+					}}>
+					Tutaj będzie panel properites
```

Comment:
Why have you reverted this polish stuff ?

Reply:
I had to revert some changes because they broke Viewport but somehow it reverted translated strings as well